cmd/ephemeral-installer-register-cluster-and-infraenv/main.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+The ephemeral-installer-register-cluster-and-infraenv is a client that runs
+in a bootable image containing all components needed to run assisted-service
+and the agent. The image is intend to be used by users to who are not currently
+supported by the IPI installer. It is also targeted for disconnected
+on-premises use.
+
+The client performs three main tasks:
+1. Reads in ZTP manifests containing information about cluster 0. The manifests
+   are either provided by users or generated by the ISO generation tool using
+   source files like install-config.yaml. The manifests are bind mounted
+   to the container running the client.
+2. Registers a cluster in assisted service using the REST-API.
+3. Registers a infraenv using the REST-API. The infraenv contains a
+   reference to the created cluster and contains any nmstate configs.
+
+A kube-api is not available in the ephemeral-installer environment. This
+is why the ZTP manifests are not applied directly to kubernetes. Instead,
+the manifests are translated into create cluster and infraenv params, and
+then posted to the REST-API to create those resources.
+
+This client is needed to register the cluster and infraenv to minimize
+user interaction and to enable automation use cases.
+
+A service running on the image, waits for the infraenv to be registered
+to assisted-service. Once registered, an infraenv id is available and
```

Comment:
`an infraenv id is available` is that still true? I thought the client forces a specific infra env ID to be used

Reply:
The ability to configure the client to use a pre-generated infra-env-id is a future enhancement.